meshcapade_addon/operators.py

```diff
@@ -0,0 +1,1634 @@
+        bpy.context.object['gender'] = gender
+        bpy.context.object['SMPL_version'] = SMPL_version
+
+        # add a texture and change the texture option based on the gender
+        # male texture if it's a male, female texture if it's female or neutral
+        if (gender == 'male'):
+            context.window_manager.smpl_tool.texture = "m"
+        else:
+            context.window_manager.smpl_tool.texture = "f"
+
+        bpy.ops.object.set_texture()
+        bpy.ops.object.reset_body_shape('EXEC_DEFAULT')
+
+        return {'FINISHED'}
+
+
+class OP_SetTexture(bpy.types.Operator):
+    bl_idname = "object.set_texture"
+    bl_label = "Set"
+    bl_description = ("Set selected texture")
+    bl_options = {'REGISTER', 'UNDO'}
+
+    @classmethod
+    def poll(cls, context):
+        try:
+            # Enable button only if in active object is mesh
+            if (context.object.type == 'MESH'):
+                return True
+            else:
+                return False
+        except: return False
+
+    def execute(self, context):
+        selection = context.window_manager.smpl_tool.texture
+        obj = bpy.context.object
+
+        # if there's no material, add one
+        if (len(obj.data.materials) == 0):
+            # the incoming name of the selected object is in the format of "SUPR-mesh-male"
+            # this line turns "SUPR-mesh-male" into "SUPR-male", which is the naming format of the materials
+            split_name = obj.name.split("-")
+            new_material_name = split_name[0] + "-" + split_name[2]
+            material = bpy.data.materials.new(name=new_material_name)
+            bpy.context.object.data.materials.append(material)
+
+        else: 
+            material = obj.data.materials[0]
+
+        # Enable the use of nodes for the material
+        material.use_nodes = True
+        node_tree = material.node_tree
+        nodes = node_tree.nodes
+
+        # if they selected the male or female texture, we add the normal map and roughness map as well
+        if selection in ('m', 'f'):
+            # Set the path to the texture files
+            albedo_map_path = os.path.join(PATH, "data", selection + "_albedo.png")
+            normal_map_path = os.path.join(PATH, "data", selection + "_normal.png")
+            roughness_map_path = os.path.join(PATH, "data", selection + "_roughness.png")
+            ao_map_path = os.path.join(PATH, "data", "ao.png")
+            thickness_map_path = os.path.join(PATH, "data", "thickness.png")
+
+            # Clear default nodes
+            for node in nodes:
+                nodes.remove(node)
+
+            # Create a new Principled BSDF node
+            principled_node = nodes.new(type="ShaderNodeBsdfPrincipled")
+            principled_node.location = 0, 0
+
+            # Add a texture node for the albedo map
+            albedo_map_node = nodes.new(type="ShaderNodeTexImage")
```

Comment:
These blockes seem like they can be factored into a function to which you pass in the node location, image path, color space, and input name? Maybe leave it for a future refactor if it's not super easy though.